code/modules/ai/ai_minds/carbon/xeno/xeno.dm

```diff
@@ -0,0 +1,113 @@
+//Generic template for application to a xeno/ mob, contains specific obstacle dealing alongside targeting only humans, xenos of a different hive and sentry turrets
+
+/datum/ai_mind/carbon/xeno
+	sidestep_prob = 100 //Kill everything
+
+//Returns a list of things we can walk to and attack to death
+/datum/ai_mind/carbon/xeno/get_targets()
+	var/list/return_result = list()
+	for(var/mob/living/carbon/human/h in cheap_get_humans_near(mob_parent, 8))
+		if(h && h.stat != DEAD)
+			return_result += h
+	var/mob/living/carbon/xenomorph/mob2 = mob_parent
+	for(var/mob/living/carbon/xenomorph/x in GLOB.alive_xeno_list)
+		if(x.hivenumber != mob2.hivenumber) //Xenomorphs not in our hive will attack as well!
+			return_result += x
+	for(var/atom/turret in GLOB.marine_turrets)
+		if(!turret.gc_destroyed && get_dist(mob_parent, turret) <= 8 && mob_parent.z == turret.z)
+			return_result += turret
+	return return_result
+
+/datum/ai_mind/carbon/xeno/do_process()
+	mob_parent.a_intent = INTENT_HARM //Killing time
```

Comment:
No need to change intent every tick.
There should be a wrapper proc for intent changing, to register properly. Something like `set_a_intent(new_intent)`